From 58c0769a47fe15ee502dddef14e2b1b077e1222e Mon Sep 17 00:00:00 2001
From: "Fabio M. De Francesco" <fmdefrancesco@gmail.com>
Date: Sat, 4 Sep 2021 12:10:16 +0200
Subject: [PATCH v1] staging: r8188eu: Remove bh spinlocks from
 _rtw_alloc_network()

Remove bh spinlocks from _rtw_alloc_network().

There is a deadlock in _rtw_alloc_network() because this function
acquires a bh spinlock that has already been taken by the callers
up in the calls chain. These callers are rtw_update_scanned_network()
and rtw_survey_event_callback().

Signed-off-by: Fabio M. De Francesco <fmdefrancesco@gmail.com>
---
 drivers/staging/r8188eu/core/rtw_mlme.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/staging/r8188eu/core/rtw_mlme.c b/drivers/staging/r8188eu/core/rtw_mlme.c
index 1115ff5d865a..0323016bdba8 100644
--- a/drivers/staging/r8188eu/core/rtw_mlme.c
+++ b/drivers/staging/r8188eu/core/rtw_mlme.c
@@ -172,8 +172,6 @@ struct	wlan_network *_rtw_alloc_network(struct	mlme_priv *pmlmepriv)/* _queue *f
 	struct __queue *free_queue = &pmlmepriv->free_bss_pool;
 	struct list_head *plist = NULL;
 
-	spin_lock_bh(&free_queue->lock);
-
 	if (list_empty(&free_queue->queue)) {
 		pnetwork = NULL;
 		goto exit;
@@ -193,8 +191,6 @@ struct	wlan_network *_rtw_alloc_network(struct	mlme_priv *pmlmepriv)/* _queue *f
 	pmlmepriv->num_of_scanned++;
 
 exit:
-	spin_unlock_bh(&free_queue->lock);
-
 	return pnetwork;
 }
 
-- 
2.33.0

